lang/src/lib.rs

```diff
@@ -154,6 +154,31 @@ pub trait ToAccountInfo<'info> {
     fn to_account_info(&self) -> AccountInfo<'info>;
 }
 
+/// Allows you to check if a given account is initialized or not
+pub trait AccountInitialized<'info>: ToAccountInfo<'info> {
+    /// Checks if account initialized by call
+    /// [`CreateAccount`](solana_program::system_instruction::SystemInstruction::CreateAccount)
+    /// or [CreateAccountWithSeed](`solana_program::system_instruction::SystemInstruction::CreateAccountWithSeed`)
+    /// solana system instructions
+    ///
+    /// According to the [documentation](https://docs.solana.com/developing/programming-model/accounts#creating),
+    /// an uninitialized account has 0 [lamports](AccountInfo::lamports) and has a [system program](solana_program::system_program) [owner](AccountInfo::owner)
+    fn initialized(&self) -> bool {
+        let account_info = self.to_account_info();
+        account_info.lamports() == 0 && system_program::ID.eq(account_info.owner)
```

Comment:
This is inconsistent with the comment, which defines this as uninitialized.

Additionally, the term `initialized` here is potentially confusing. In Anchor, initialized suggests, data space is allocated, the program is the owner, and an account discriminator is assigned (though there's some minor cleanup we need to do internally to actually make this the case).

Reply:
1. Agree, fixed.
2. Any ideas? `exists` + `not_exists` or `created` + `not_created`?

Reply:
Probably can remove the trait since it's only used in one non deprecated type.

Reply:
As I said above, I wanted to keep the documentation in rustdoc more, but I do not insist. ✌️